### PR TITLE
Create screen share view marged with camera view

### PR DESCRIPTION
**Pull request by Karolk99**
closes also: https://membraneframework.atlassian.net/jira/software/c/projects/LIVE/boards/11?modal=detail&selectedIssue=LIVE-79

### CODE REVIEW

File: assets/js/utils/rtcUtils.ts [hunk 1]
```diff
@@ -184,3 +159,124 @@ export const leaveWebrtc = (webrtc: MembraneWebRTC, clientName: string, webrtcCh
...
+
+  mergedStream.getTracks().forEach((track) => {
+    addOrReplaceTrack(clientName, track, playerCallback);
+  });
+
+  const newTrack = findTrackByType(clientName, "video");
+
+  if (!webrtc || !newTrack) return;
+  if (sourceIds["video"]) webrtc.replaceTrack(sourceIds["video"], newTrack);
+  else sourceIds["video"] = webrtc.addTrack(newTrack, presenterStreams[clientName]);
```
Comment (Rados13) [inline assets/js/utils/rtcUtils.ts:179]: Could we simply use here `addOrReplaceTrack`?

File: assets/js/utils/rtcUtils.ts [hunk 2]
```diff
@@ -6,30 +6,27 @@ export type Sources = {
...
-  const curTrack = findTrackByType(name, track.kind as SourceType);
-  if (curTrack) {
-    curTrack.stop();
-    presenterStreams[name].removeTrack(curTrack);
-  }
-  presenterStreams[name].addTrack(track);
-  playerCallback(track.kind as SourceType); // to attach MediaStream to HTMLVideoElement object in DOM
+const VIDEO_CONSTRAINTS: MediaStreamConstraints = {
+  audio: false,
+  video: { width: 1280, height: 720, frameRate: FRAME_RATE },
```
Comment (Rados13) [inline assets/js/utils/rtcUtils.ts:19]: I think we should move it to a separate file like `const.ts`

File: assets/js/utils/rtcUtils.ts [hunk 3]
```diff
@@ -184,3 +159,124 @@ export const leaveWebrtc = (webrtc: MembraneWebRTC, clientName: string, webrtcCh
...
+    : console.error("CanvasCtx is null", canvasCtx);
+
+  return canvasElement.captureStream(FRAME_RATE);
+};
+
+const attachToDOM = async (id: string, stream: MediaStream) => {
+  let videoElem = document.createElement("video");
+  videoElem.id = id;
+  videoElem.width = 1280;
+  videoElem.height = 720;
```
Comment (Rados13) [inline assets/js/utils/rtcUtils.ts:205]: I think this `videoElem` should have the same dimensions as in this in `VIDEO_CONSTRAINTS`

**Review by Rados13 — CHANGES_REQUESTED** [review]
In addition, I would move code that works on canvas and DOM elements to another file so that `rtcUtils.ts` handle only rtc-related things.

File: assets/js/utils/rtcUtils.ts (hunk 1 above)
Comment (Karolk99) [reply assets/js/utils/rtcUtils.ts:179]: Those are two different functionalities

File: assets/js/utils/rtcUtils.ts [hunk 4]
```diff
@@ -64,6 +62,7 @@ export const setSourceById = async (
     localStream = await navigator.mediaDevices.getUserMedia(getConstraint(constraint, deviceId));
 
     localStream.getTracks().forEach((track) => {
+      console.log(track);
```
Comment (Rados13) [inline assets/js/utils/rtcUtils.ts:65]: Do we need this?

File: assets/js/utils/canvasUtils.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
+import { FRAME_RATE, SCREEN_CONSTRAINTS, CANVAS_HEIGHT, CANVAS_WIDTH } from "./const";
+import type { MergedScreenRef } from "./rtcUtils";
+
+export const getMergedTracks = async (
+  mergedScreenRef: MergedScreenRef,
+  presenterStream: MediaStream
+) => {
+  if (mergedScreenRef.refreshId) stopRefreshingAndRemovePreviousScreen(mergedScreenRef);
```
Comment (Rados13) [inline assets/js/utils/canvasUtils.ts:8]: Could we make it like this?
```suggestion
  mergedScreenRef.refreshId && stopRefreshingAndRemovePreviousScreen(mergedScreenRef);
```

**Review by Rados13 — APPROVED** [review]

**Pull request merged**